modules/contact/src/actions/ContactAction.C

```diff
@@ -119,6 +120,15 @@ ContactAction::ContactAction(const InputParameters & params)
     if (_model == "glued")
       paramError("model", "The 'mortar' formulation does not support glued contact (yet)");
   }
+
+  // Make the Dirac kernel deprecated as an initial step for its final removal.
+  MooseEnum makeDiracDeprecated = params.get<MooseEnum>("system");
+  makeDiracDeprecated.deprecate("DiracKernel", "Constraint");
+
+  if (_system == "DiracKernel")
+    mooseDeprecated(
+        "The Dirac kernel system for contact mechanics is deprecated and will be finally "
+        "removed on April 1, 2020. The recommended option is the constraint system.\n");
```

Comment:
Minor suggested rewording to make this a little more clear to users:

The DiracKernel-based system for mechanical contact enforcement is deprecated, and will be removed on April 1, 2020.  It is being replaced by the Constraint-based system, which is selected by setting 'system=Constraint'.